core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -253,6 +253,8 @@ class RequestSendThread(val controllerId: Int,
 
   private var firstUpdateMetadataWithPartitionsSent = false
 
+  private var fullLeaderAndIsrSent = false
```

Comment:
Thanks for the suggestions. I'll adopt the 1st one while leaving out the remaining two since
1. the variable names are already self-explanatory
2. for the UMR, the 1st request is actually not the one we want because it only contains info about live brokers, and only the 2nd request contains info about partitions.